src/Microsoft.AspNetCore.Routing/RouteValuesBasedEndpointFinder.cs

```diff
@@ -78,15 +76,16 @@ private void BuildOutboundMatches()
         {
             // Refresh the matches in the case where a datasource's endpoints changes. The following is OK to do
             // as refresh of new endpoints happens within a lock and also these fields are not publicly accessible.
-            var (allMatches, namedMatches) = GetOutboundMatches();
-            _namedMatches = GetNamedMatches(namedMatches);
+            var (allMatches, namedMatchResults) = GetOutboundMatches();
+            _namedMatchResults = namedMatchResults;
             _allMatchesLinkGenerationTree = new LinkGenerationDecisionTree(allMatches.ToArray());
         }
 
-        protected virtual (IEnumerable<OutboundMatch>, IDictionary<string, List<OutboundMatch>>) GetOutboundMatches()
+        protected virtual (IEnumerable<OutboundMatch>, IDictionary<string, List<OutboundMatchResult>>) GetOutboundMatches()
```

Comment:
Is there a good reason for this to be protected? used in tests? a comment would help